modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java

```diff
@@ -748,7 +743,18 @@ private GridCoverage2D renderCoverage(final GridCoverage2D gridCoverage,
         }
     }
 
-
+	private RenderedImage getImageFromParentCoverage(GridCoverage2D parentCoverage) {
+		if (parentCoverage == null) {
+			return null;
+		}
+		RenderedImage ri = parentCoverage.getRenderedImage();
+		if (ri != null) {
+			PlanarImage pi = PlanarImage.wrapRenderedImage(ri);
```

Comment:
This already checks if ri is a planar and avoids double wrapping, right?

Reply:
Yep. wrapRenderedImage avoid double wrapping by simply returning the input object if already a PlanarImage.